packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx

```diff
@@ -0,0 +1,688 @@
+        googlepay: '/img/payment-providers/google-pay.png',
+        humm: '/img/payment-providers/humm-checkout-header.png',
+        klarna: '/img/payment-providers/klarna-header.png',
+        laybuy: '/img/payment-providers/laybuy-checkout-header.png',
+        masterpass: 'https://masterpass.com/dyn/img/acc/global/mp_mark_hor_blk.svg',
+        paypal: '/img/payment-providers/paypalpaymentsprouk.png',
+        quadpay: '/img/payment-providers/quadpay.png',
+        sezzle: '/img/payment-providers/sezzle-checkout-header.png',
+        zip: '/img/payment-providers/zip.png',
+        paypalcommerce: '/img/payment-providers/paypal_commerce_logo.svg',
+    };
+
+    beforeEach(() => {
+        defaultProps = {
+            formValues: {
+                ccExpiry: '10 / 22',
+                ccName: 'Good Shopper',
+                ccNumber: '4111 1111 1111 1111',
+                paymentProviderRadio: getPaymentMethod().id,
+            },
+            method: getPaymentMethod(),
+        };
+
+        config = getStoreConfig();
+        checkoutService = createCheckoutService();
+        localeContext = createLocaleContext(config);
+        jest.spyOn(checkoutService.getState().data, 'getConfig').mockReturnValue(config);
+
+        PaymentMethodTitleTest = ({ formValues, ...props }) => (
+            <CheckoutProvider checkoutService={checkoutService}>
+                <LocaleContext.Provider value={localeContext}>
+                    <Formik initialValues={formValues} onSubmit={noop}>
+                        <PaymentMethodTitle {...props} />
+                    </Formik>
+                </LocaleContext.Provider>
+            </CheckoutProvider>
+        );
+    });
+
+    it('renders name of payment method if it does not have logo', () => {
+        render(<PaymentMethodTitleTest {...defaultProps} />);
+
+        expect(screen.getByRole('heading', { name: defaultProps.method.config.displayName })).toBeInTheDocument();
+    });
+
+    it('renders name of payment method if it is credit card method', () => {
+        render(
+            <PaymentMethodTitleTest
+                {...defaultProps}
+                method={{
+                    ...defaultProps.method,
+                    method: 'credit-card',
+                }}
+            />,
+        );
+
+        expect(screen.getByRole('heading', { name: defaultProps.method.config.displayName })).toBeInTheDocument();
+    });
+
+    it('renders logo based on their method type', () => {
+        render(
+            <PaymentMethodTitleTest
+                {...defaultProps}
+                method={{
+                    ...defaultProps.method,
+                    method: PaymentMethodType.ApplePay,
+                }}
+            />,
+        );
+
+        const expectedPath = /^http/.test(LOGO_PATHS[PaymentMethodType.ApplePay])
+            ? LOGO_PATHS[PaymentMethodType.ApplePay]
+            : `${config.cdnPath}${LOGO_PATHS[PaymentMethodType.ApplePay]}`;
```

Comment:
Is it possible that `expectedPath` here can be different on different test runs?
If not, we should be able to remove the `if` here.